R/data.table.R

```diff
@@ -483,6 +483,10 @@ replace_dot_alias <- function(e) {
       i = as.data.table(i)
     }
     if (is.data.table(i)) {
+      if (!haskey(x) && missing(on) && length(common_names <- intersect(names(x), names(i)))) {
+        if (verbose) cat("Joining but 'x' has no key and 'on' is missing, natural join using common columns\n")
```

Comment:
this message should name the columns

Reply:
I wanted to avoid that just in case there would be many columns, then there is no point to print them.